### ISSUE TITLE
HDF5-Mode: Members of linked FTIT_H5Group list out of scope 

### ISSUE THREAD

**Issue by kellekai** [leobago/fti#181]
imagine scenario:  
```C
add_group () {
  FTIT_H5Group foo;
  FTI_InitGroup( &foo, "foo", NULL );
}
```  
`foo` is local to `add_group()`.  
in `FTI_InitGroup(FTIT_H5Group* h5group, char* name, FTIT_H5Group* parent)` we do:  
`parent->children[parent->childrenNo] = h5group;`
but the pointer `h5group` appended to the parent group is not valid outside of `add_group`.  
  
possible soulution is a `malloc` inside `FTI_InitGroup`. Or passing a pointer-to-pointer:  
```C
FTI_InitGroup(FTIT_H5Group** h5group, char* name, FTIT_H5Group* parent)
```
and
```C
add_group () {
  FTIT_H5Group *foo;
  FTI_InitGroup( &foo, "foo", NULL );
}
```

**Comment by ksiero** [comment]
This problems also apply to FTIT_Type if using HDF5 format

**Comment by ksiero** [comment]
Solved in #183.
